### PR TITLE
PAC/BTI enablement in TS

### PR DESCRIPTION
**Pull request by gabor-toth-arm**
Should be merged together with or after the listed PR-s:
- https://github.com/OP-TEE/optee_os/pull/6945
- https://github.com/OP-TEE/optee_os/pull/6988

### CODE REVIEW

File: fvp-psa-sp.mk [hunk 1]
```diff
@@ -13,12 +13,15 @@ TS_UEFI_INTERNAL_CRYPTO	?= n
 SP_PACKAGING_METHOD		?= embedded
 SPMC_TESTS			?= n
 SPMC_AT_EL			?= 1
-TS_BTI_ENABLED			?= unset
```
Comment (jforissier) [inline fvp-psa-sp.mk]: It doesn't make much sense to introduce `TS_BTI_ENABLED` in the parent commit only to remove it in this one. Please squash the two commits into "fvp: add PAC and BTI support".

Comment (gabor-toth-arm) [reply fvp-psa-sp.mk]: Yes, you are right. Squashed them into a single commit.

File: fvp-psa-sp.mk [hunk 2]
```diff
@@ -80,6 +89,32 @@ OPTEE_OS_COMMON_EXTRA_FLAGS += \
...
+# If branch protection is unset, do not pass it
+ifeq ($(filter $(TS_BRANCH_PROTECTION),unset),)
+TF_A_FLAGS              += BRANCH_PROTECTION=$(TS_BRANCH_PROTECTION)
+TS_APP_COMMON_FLAGS	+= -DBRANCH_PROTECTION=$(TS_BRANCH_PROTECTION)
+SP_COMMON_FLAGS		+= -DBRANCH_PROTECTION=$(TS_BRANCH_PROTECTION)
+endif
+
+# Branch Target Identification enablement
+ifneq ($(filter $(TS_BRANCH_PROTECTION),1 4),)
+	OPTEE_OS_COMMON_EXTRA_FLAGS += CFG_CORE_BTI=y
```
Comment (etienne-lms) [inline fvp-psa-sp.mk:101]: It's better to avoid tabs on variable assignment to prevent confusions with target's commands.

File: fvp-psa-sp.mk [hunk 3]
```diff
@@ -14,6 +14,15 @@ SP_PACKAGING_METHOD		?= embedded
...
 SPMC_AT_EL			?= 1
 
+# Behaves a similar way like BRANCH_PROTECTION in TF-A:
+# unset: Default value. mbranch-protection flag is not provided
+# 0: Turns off all types of branch protection
+# 1: Enables all types of branch protection features
+# 2: Return address signing to its standard level
+# 3: Extend the signing to include leaf functions
+# 4: Turn on branch target identification mechanism
+TS_BRANCH_PROTECTION		?= unset
```
Comment (etienne-lms) [inline fvp-psa-sp.mk:24]: Should there be a sanity test on the value assigned to `TS_BRANCH_PROTECTION` (1|..|4|unset) ?

**Comment by gabor-toth-arm** [comment]
Hi @jforissier @etienne-lms !

Could we merge this commit if there are no more findings or is there anything else I should correct?

**Comment by jforissier** [comment]
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@foss.st.com>` with s/BRANCH_PROTECTION/TS_BRANCH_PROTECTION/ in commit message.

**Comment by gabor-toth-arm** [comment]
Hi @jenswi-linaro !

Could you take a look at this change and merge it?

Thank you!

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jenswi-linaro** [comment]
This is ready to be merged.

**Pull request merged**